### PR TITLE
[Migrate]: Pod level pipeline from self hosted runner to shared runner

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: udit <udit@chaosnative.com>

## Details 

- Shift pod level pipeline clusters from kubeadm on ec2 to a transient kind/k3s cluster
- We can use a shared runner also in place of the self-hosted runner.
- The pipeline has a well-defined debug step to catch any test failure.

#### Issue:
- https://github.com/litmuschaos/litmus-e2e/issues/299 

#### Screenshot

image

### CODE REVIEW

**Review by oumkale — APPROVED** [review]

File: .github/workflows/pod-level-pipeline.yml [hunk 1]
```diff
@@ -21,93 +23,140 @@ on:
...
+      OPERATOR_IMAGE: "${{ github.event.inputs.operatorImage }}"
+      OPERATOR_NAME: "${{ github.event.inputs.operatorName }}"
+      RUNNER_IMAGE: "${{ github.event.inputs.runnerImage }}"
+      IMAGE_PULL_POLICY: "${{ github.event.inputs.imagePullPolicy }}"        
+      GO_EXPERIMENT_IMAGE: "${{ github.event.inputs.goExperimentImage }}"
+      LIB_IMAGE: "${{ github.event.inputs.libImage }}"
+      EXPERIMENT_IMAGE_PULL_POLICY: "${{ github.event.inputs.experimentImagePullPolicy }}"
+      CHAOS_NAMESPACE: "${{ github.event.inputs.chaosNamespace }}"
+      APP_NS: "${{ github.event.inputs.chaosNamespace }}"
+      CHAOS_SERVICE_ACCOUNT: "${{ github.event.inputs.chaosServiceAccount }}"
```
Comment (Jonsy13) [inline .github/workflows/pod-level-pipeline.yml:42]: This `chaosServiceAccount` input is not available in inputs array

File: .github/workflows/pod-level-pipeline.yml [hunk 2]
```diff
@@ -21,93 +23,140 @@ on:
...
-    runs-on: [self-hosted, pod-level]
-    ## Changing the working directory to image path we provided
-    ## As the default path is repository checkout path.
+
+  Network_Chaos_Test:
+    if: always()
+    runs-on: ubuntu-latest
+    env:
+      OPERATOR_IMAGE: "${{ github.event.inputs.operatorImage }}"
+      OPERATOR_NAME: "${{ github.event.inputs.operatorName }}"
```
Comment (Jonsy13) [inline .github/workflows/pod-level-pipeline.yml:34]: This `operatorName` input is not available in inputs array

File: .github/workflows/pod-level-pipeline.yml [hunk 3]
```diff
@@ -21,93 +23,140 @@ on:
...
+      RUNNER_IMAGE: "${{ github.event.inputs.runnerImage }}"
+      IMAGE_PULL_POLICY: "${{ github.event.inputs.imagePullPolicy }}"        
+      GO_EXPERIMENT_IMAGE: "${{ github.event.inputs.goExperimentImage }}"
+      LIB_IMAGE: "${{ github.event.inputs.libImage }}"
+      EXPERIMENT_IMAGE_PULL_POLICY: "${{ github.event.inputs.experimentImagePullPolicy }}"
+      CHAOS_NAMESPACE: "${{ github.event.inputs.chaosNamespace }}"
+      APP_NS: "${{ github.event.inputs.chaosNamespace }}"
```
Comment (Jonsy13) [inline .github/workflows/pod-level-pipeline.yml:127]: Same comments as above for inputs.

File: .github/workflows/scheduled-pod-level-pipeline.yml [hunk 4]
```diff
@@ -4,80 +4,123 @@ on:
...
+    
     steps:
+    
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.16'
+
+      - uses: actions/checkout@v2
+        with:
+          repository: '${{ github.event.inputs.e2eTestRepo }}'
```
Comment (Jonsy13) [inline .github/workflows/scheduled-pod-level-pipeline.yml:27]: These both inputs `e2eTestRepo` & `e2eTestBranch` won't be available in scheduled pipelines

File: .github/workflows/scheduled-pod-level-pipeline.yml [hunk 5]
```diff
@@ -4,80 +4,123 @@ on:
...
-        if: ${{ always() }}
-        run: make container-kill
+    
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.16'
+
+      - uses: actions/checkout@v2
+        with:
+          repository: '${{ github.event.inputs.e2eTestRepo }}'
```
Comment (Jonsy13) [inline .github/workflows/scheduled-pod-level-pipeline.yml:105]: Same as above

File: .github/workflows/scheduled-pod-level-pipeline.yml [hunk 6]
```diff
@@ -109,87 +144,406 @@ jobs:
...
+    
+    steps:
+    
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.16'
+
+      - uses: actions/checkout@v2
+        with:
+          repository: '${{ github.event.inputs.e2eTestRepo }}'
```
Comment (Jonsy13) [inline .github/workflows/scheduled-pod-level-pipeline.yml:192]: Same as above

File: .github/workflows/scheduled-pod-level-pipeline.yml [hunk 7]
```diff
@@ -109,87 +144,406 @@ jobs:
...
```
Comment (Jonsy13) [inline .github/workflows/scheduled-pod-level-pipeline.yml:258]: Same as above

File: .github/workflows/scheduled-pod-level-pipeline.yml [hunk 8]
```diff
@@ -109,87 +144,406 @@ jobs:
...
+    
     steps:
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.16'
+
+      - uses: actions/checkout@v2
+        with:
+          repository: '${{ github.event.inputs.e2eTestRepo }}'
```
Comment (Jonsy13) [inline .github/workflows/scheduled-pod-level-pipeline.yml:326]: Same as above

File: .github/workflows/scheduled-pod-level-pipeline.yml [hunk 9]
```diff
@@ -109,87 +144,406 @@ jobs:
...
```
Comment (Jonsy13) [inline .github/workflows/scheduled-pod-level-pipeline.yml:403]: Same as above

File: .github/workflows/scheduled-pod-level-pipeline.yml [hunk 10]
```diff
@@ -109,87 +144,406 @@ jobs:
...
```
Comment (Jonsy13) [inline .github/workflows/scheduled-pod-level-pipeline.yml:482]: Same as above

File: .github/workflows/pod-level-pipeline.yml [hunk 11]
```diff
@@ -116,61 +165,241 @@ jobs:
...
+      - name: Deleting K3S cluster
+        if: always()
+        run: /usr/local/bin/k3s-uninstall.sh   
+
+  Container_Kill_Test:
+    if: always()
+    runs-on: ubuntu-latest
+    env:
+      OPERATOR_IMAGE: "${{ github.event.inputs.operatorImage }}"
+      OPERATOR_NAME: "${{ github.event.inputs.operatorName }}"
```
Comment (Jonsy13) [inline .github/workflows/pod-level-pipeline.yml:215]: Same as above

File: .github/workflows/pod-level-pipeline.yml [hunk 12]
```diff
@@ -116,61 +165,241 @@ jobs:
...
+      - name: Deleting K3S cluster
+        if: always()
+        run: /usr/local/bin/k3s-uninstall.sh        
+
+  Pod_Delete_Test:
+    if: always()
+    runs-on: ubuntu-latest
+    env:
+      OPERATOR_IMAGE: "${{ github.event.inputs.operatorImage }}"
+      OPERATOR_NAME: "${{ github.event.inputs.operatorName }}"
```
Comment (Jonsy13) [inline .github/workflows/pod-level-pipeline.yml:289]: Same as above

File: .github/workflows/pod-level-pipeline.yml [hunk 13]
```diff
@@ -116,61 +165,241 @@ jobs:
...
-        KUBECONFIG: /root/.kube/config
-
-    runs-on: [self-hosted, pod-level-exp-tunable]
+    runs-on: ubuntu-latest
+    env:
+      GO_EXPERIMENT_IMAGE: "${{ github.event.inputs.goExperimentImage }}"
+      LIB_IMAGE: "${{ github.event.inputs.libImage }}"
+      EXPERIMENT_IMAGE_PULL_POLICY: "${{ github.event.inputs.experimentImagePullPolicy }}"
+      CHAOS_NAMESPACE: "${{ github.event.inputs.expTunnableNS }}"
+      CHAOS_SERVICE_ACCOUNT: "${{ github.event.inputs.chaosServiceAccount }}"
```
Comment (Jonsy13) [inline .github/workflows/pod-level-pipeline.yml:368]: Same as above

File: .github/workflows/pod-level-pipeline.yml [hunk 14]
```diff
@@ -181,55 +410,206 @@ jobs:
...
-        KUBECONFIG: /root/.kube/config
-
-    runs-on: [self-hosted, pod-level]
+    runs-on: ubuntu-latest
+    env:
+      GO_EXPERIMENT_IMAGE: "${{ github.event.inputs.goExperimentImage }}"
+      LIB_IMAGE: "${{ github.event.inputs.libImage }}"
+      EXPERIMENT_IMAGE_PULL_POLICY: "${{ github.event.inputs.experimentImagePullPolicy }}"
+      CHAOS_NAMESPACE: "${{ github.event.inputs.expTunnableNS }}"
+      CHAOS_SERVICE_ACCOUNT: "${{ github.event.inputs.chaosServiceAccount }}"
```
Comment (Jonsy13) [inline .github/workflows/pod-level-pipeline.yml:454]: Same as above

File: .github/workflows/pod-level-pipeline.yml [hunk 15]
```diff
@@ -181,55 +410,206 @@ jobs:
...
-        POD_LEVEL: "true"
-
-    runs-on: [self-hosted, pod-level]
+    runs-on: ubuntu-latest
+    env:
+      GO_EXPERIMENT_IMAGE: "${{ github.event.inputs.goExperimentImage }}"
+      LIB_IMAGE: "${{ github.event.inputs.libImage }}"
+      EXPERIMENT_IMAGE_PULL_POLICY: "${{ github.event.inputs.experimentImagePullPolicy }}"
+      CHAOS_NAMESPACE: "${{ github.event.inputs.expTunnableNS }}"
+      CHAOS_SERVICE_ACCOUNT: "${{ github.event.inputs.chaosServiceAccount }}"
```
Comment (Jonsy13) [inline .github/workflows/pod-level-pipeline.yml:542]: Same as above

**Review by Jonsy13 — CHANGES_REQUESTED** [review]

File: .github/workflows/pod-level-pipeline.yml (hunk 13 above)
Comment (Jonsy13) [reply .github/workflows/pod-level-pipeline.yml:364]: PTAL `EXPERIMENT_IMAGE_PULL_POLICY: "${{ github.event.inputs.experimentImagePullPolicy }}"`, This is also not present in inputs array. Check in other places too.

**Review by Jonsy13 — APPROVED** [review]

**Pull request merged**